### ISSUE TITLE
assert.async breaks tests in node

### ISSUE THREAD

**Issue by getify** [qunitjs/qunit#1129]
@leobalter

I'm using QUnit 2.1.1.

My minimal test case:

```js
// tmp.js
var QUnit = require( "qunitjs" );

QUnit.done( function done(results){
	console.log( results );
} );

QUnit.test( "temp", function test(assert){
	var done = assert.async();
	setTimeout( done, 500 );
} );

QUnit.start();
```

```
$] node --version
v7.7.1
$] node tmp.js
{ failed: 1, passed: 0, total: 1, runtime: 5 }
{ failed: 1, passed: 0, total: 1, runtime: 523 }
```

As you can see, the `Qunit.done(..)` callback is called **twice**, once right away, and once 500 ms later. :(

**Comment by getify** [comment]
BTW, I just updated to Node 7.7.4 and the bug persists, so it's not related to my node version.

**Comment by trentmwillis** [comment]
The latest version is 2.2.1, did you mean this is fixed in 2.2.2?

**Comment by leobalter** [comment]
> Update: I just tried pulling down ?-pre from github...

That's probably the reason I was not able to reproduce this bug when we talked.

It seems like a regression already fixed by some other change. We can do a git blame to find it, but at this point I believe it's better to release a new version, which I guess we can do it this week.

**Comment by getify** [comment]
@trentmwillis sorry, yes, I meant 2.2.2 (vs 2.2.1)

@leobalter it does seem though like you'd want to get a test in the suite that fails in 2.2.1 so this regression is caught if it ever happens again. :)

**Comment by leobalter** [comment]
sure thing. I'll do it.

**Comment by getify** [comment]
OK, sorry for my confusion.

Just realized I had indeed been running on 2.**1.1** (but thinking I was on latest 2.**2.1**). So my bug was only on 2.**1.1** and fixed sometime after. I just verified that in both 2.**2.1** and 2.**2.2**(-pre), the bug is not present.

Maybe still want to put a test in, but sorry for my false alarm here. Closing.

**Comment by trentmwillis** [comment]
I think the root cause was likely the same as: https://github.com/qunitjs/qunit/pull/1103. Which would explain why it is fixed in 2.2. That said, I will investigate this to verify.
